code/game/machinery/reagents/pump.dm

```diff
@@ -186,66 +193,78 @@
 		cell.update_icon()
 
 		cell = null
-		on = 0
+		turn_off(TRUE)
 		set_light(0)
-		to_chat(user, "<span class='notice'>You remove the power cell.</span>")
+		to_chat(user, SPAN_NOTICE("You remove the power cell."))
 		update_icon()
 		return
 
 	if(on)
 		turn_off(TRUE)
 	else if(anchored)
-		if(!turn_on(1))
-			to_chat(user, "<span class='notice'>You try to turn on \the [src] but it does not work.</span>")
+		if(!turn_on(TRUE))
+			to_chat(user, SPAN_NOTICE("You try to turn on \the [src] but nothing happens."))
 
 	update_icon()
 
 /obj/machinery/pump/attackby(obj/item/weapon/W as obj, mob/user as mob)
 	if(W.is_screwdriver())
 		if(!open)
 			if(unlocked)
-				unlocked = 0
-				to_chat(user, "<span class='notice'>You screw the battery panel in place.</span>")
+				unlocked = FALSE
+				to_chat(user, SPAN_NOTICE("You screw the battery panel in place."))
 			else
-				unlocked = 1
-				to_chat(user, "<span class='notice'>You unscrew the battery panel.</span>")
+				unlocked = TRUE
+				to_chat(user, SPAN_NOTICE("You unscrew the battery panel."))
+		else if(!cell)
+			default_deconstruction_screwdriver(user, W)
+			update_icon()
+			return
 
 	else if(W.is_crowbar())
 		if(unlocked)
 			if(open)
-				open = 0
-				overlays = null
-				to_chat(user, "<span class='notice'>You crowbar the battery panel in place.</span>")
+				if(default_deconstruction_crowbar(user, W))
+					return
+
+				open = FALSE
+				to_chat(user, SPAN_NOTICE("You pry the battery panel into place."))
 			else
 				if(unlocked)
-					open = 1
-					to_chat(user, "<span class='notice'>You remove the battery panel.</span>")
+					open = TRUE
+					to_chat(user, SPAN_NOTICE("You remove the battery panel."))
 
 	else if(W.is_wrench())
 		if(on)
-			to_chat(user, "<span class='notice'>\The [src] is active. Turn it off before trying to move it!</span>")
+			to_chat(user, SPAN_NOTICE("\The [src] is active. Turn it off before trying to move it!"))
 			return
 		default_unfasten_wrench(user, W, 2 SECONDS)
 
 	else if(istype(W, /obj/item/weapon/cell))
 		if(open)
+			if(panel_open)
+				to_chat(user, SPAN_NOTICE("The cell can't be connected directly to the machine's wires. Replace the maintenance panel."))
 			if(cell)
-				to_chat(user, "<span class='notice'>There is a power cell already installed.</span>")
+				to_chat(user, SPAN_NOTICE("There is a power cell already installed."))
 			else
-				user.drop_item()
+				user.drop_from_inventory(W)
 				W.loc = src
 				cell = W
-				to_chat(user, "<span class='notice'>You insert the power cell.</span>")
+				to_chat(user, SPAN_NOTICE("You insert the power cell."))
 	else
-		..()
+		. = ..()
 	RefreshParts()
 	update_icon()
 
 /obj/machinery/pump/proc/handle_pumping()
 	var/turf/T = get_turf(src)
 
-	if(istype(T, /turf/simulated/floor/water))
-		cell.use(use * CELLRATE / efficiency)
```

Comment:
You don't seem to be replacing this cell.use() call anywhere?

Reply:
checked_use calls use if the cell actually has charge for it.